WPMedia/fusion-news-theme-blocks
# Themes 1198 social share bar hover styles

**This PR goes with an `arc-themes-feature-pack` PR (you can find it [here](https://github.com/WPMedia/arc-themes-feature-pack/pull/419)) and an `arc-themes-components` PR (you can find it [here](https://github.com/WPMedia/arc-themes-components/pull/191))**

## Description

The goal of this ticket is to update the hover style of the `share-bar-block` based on the [hover states audit](https://paper.dropbox.com/doc/Themes-hover-states-audit--B6_VBNBSwmk32w6n~v6al4KOAg-JyHvxHuZttlIfj5BzjTI3). I made the following changes:
* `share-bar-block/themes/news.json`:
    * Gave `icon` elements a uniform fill color. The social icons are now black (`$color-primary` in [this design guide](https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?type=design&node-id=7053-58820&mode=design&t=BNqnKC1OplzIcwN6-0))
    * Set the color of the social icons to be `#384355` (`$color-primary-hover` in the [design guide](https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?type=design&node-id=7053-58820&mode=design&t=BNqnKC1OplzIcwN6-0))
    * Removed the original fill colors for the social icons. Originally, all of the social icons had fill colors that matched the primary color of the respective social media
* Updated the news styling in `./storybook/themes` to reflected the changes made to the block's `themes` directory
* To add a hover selector to the share bar, I had to modify the properties of the `icon` component in `arc-themes-components`
    * This has a separate PR that can be found [here](https://github.com/WPMedia/arc-themes-components/pull/191)
    * The changes are in `_index.scss` in the `icon` component
    * This is needed to run this code    
* As a result of these changes, some changes were also automatically made to the `site-styles` directory in `arc-themes-feature-pack`
    * You can find a link to that PR [here](https://github.com/WPMedia/arc-themes-feature-pack/pull/419)
    * These are needed to run this code

## Jira Ticket

[THEMES-1198](https://arcpublishing.atlassian.net/browse/THEMES-1198)

## Acceptance Criteria

For the social icons:
1. rest: `$color-primary` (`#000000`)
2. hover: `$color-primary-hover` (`#384355`)

## Test Steps

1. Checkout this branch `git checkout THEMES-1198-social-share-bar-hover-styles-2`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/share-bar-block`
3. Open your local PageBuilder instance and open the 'All Blocks-Test-TA-Sandbox' page
    * The Website type doesn't matter
    * Any other page with a social share bar block works too
4. Verify that the social icons are black, and verify each icon turns `#384355` when the mouse hovers over it

## Effect Of Changes

### Before

https://github.com/WPMedia/arc-themes-blocks/assets/103296586/113cc4a4-c363-433b-bc05-79c2d5668abf

### After

https://github.com/WPMedia/arc-themes-blocks/assets/103296586/776aec34-f9d2-4b72-b006-9aac67b28953

## Dependencies or Side Effects

This PR is dependent on some changes to `arc-themes-components`. The PR is [here](https://github.com/WPMedia/arc-themes-components/pull/191). It's also dependent on some changes to `arc-themes-feature-pack`. The PR is [here](https://github.com/WPMedia/arc-themes-feature-pack/pull/419).

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-1198]: https://arcpublishing.atlassian.net/browse/THEMES-1198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
